rust_src/src/marker.rs

```diff
@@ -102,4 +107,63 @@ pub fn set_point_from_marker(marker: LispMarkerRef) {
     unsafe { set_point_both(charpos, bytepos) };
 }
 
+/// Return insertion type of MARKER: t if it stays after inserted text.
+/// The value nil means the marker stays before text inserted there.
+#[lisp_fn]
+pub fn marker_insertion_type(marker: LispMarkerRef) -> bool {
+    marker.insertion_type()
+}
+
+/// Set the insertion-type of MARKER to TYPE.
+/// If ITYPE is non-nil, it means the marker advances when you insert text at it.
+/// If ITYPE is nil, it means the marker stays behind when you insert text at it.
+#[lisp_fn]
+pub fn set_marker_insertion_type(mut marker: LispMarkerRef, itype: LispObject) -> LispObject {
+    unsafe { mset_insertion_type(marker.as_mut(), itype.is_not_nil() as BoolBF) };
+    itype
+}
+
+/// Position MARKER before character number POSITION in BUFFER.
+/// If BUFFER is omitted or nil, it defaults to the current buffer.  If
+/// POSITION is nil, makes marker point nowhere so it no longer slows down
+/// editing in any buffer.  Returns MARKER.
+#[lisp_fn(min = "2")]
+pub fn set_marker(marker: LispObject, position: LispObject, buffer: LispObject) -> LispObject {
+    unsafe {
+        LispObject::from_raw(set_marker_internal(
+            marker.to_raw(),
+            position.to_raw(),
+            buffer.to_raw(),
+            false,
+        ))
+    }
+}
+
+/// Return a new marker pointing at the same place as MARKER.
+/// If argument is a number, makes a new marker pointing
+/// at that position in the current buffer.
+/// If MARKER is not specified, the new marker does not point anywhere.
+/// The optional argument ITYPE specifies the insertion type of the new marker;
+/// see `marker-insertion-type'.
+#[lisp_fn(min = "0")]
+pub fn copy_marker(marker: LispObject, itype: LispObject) -> LispObject {
+    if marker.is_not_nil() {
+        marker.as_fixnum_coerce_marker_or_error();
+    }
+    let new = unsafe { LispObject::from_raw(Fmake_marker()) };
+    let buffer_or_nil = if marker.is_marker() {
+        marker_buffer(marker)
```

Comment:
This no longer correct. `marker_buffer` now returns a `Option<Marker>`.

I think this should be:

buffer_or_nil = marker.as_marker().and_then(|m| m.buffer()).unwrap_or(LispObject::constant_nil(), |b| b.as_lisp_obj())

With proper formatting of course :-)

Reply:
Yeah, I have changed it. I should have made this one PR, sorry :)

Reply:
No wories, it is easy to fix.